satellite/repair/repairer/segments.go

```diff
@@ -95,21 +95,21 @@ func (repairer *SegmentRepairer) Repair(ctx context.Context, path storj.Path) (e
 	pieces := pointer.GetRemote().GetRemotePieces()
 	missingPieces, err := repairer.cache.GetMissingPieces(ctx, pieces)
 	if err != nil {
-		return Error.New("error getting missing pieces %s", err)
+		return false, Error.New("error getting missing pieces %s", err)
 	}
 
 	numHealthy := len(pieces) - len(missingPieces)
 	// irreparable piece, we need k+1 to detect corrupted pieces
 	if int32(numHealthy) < pointer.Remote.Redundancy.MinReq+1 {
 		mon.Meter("repair_nodes_unavailable").Mark(1)
-		return Error.Wrap(IrreparableError.New("segment %v cannot be repaired: only %d healthy pieces, %d required", path, numHealthy, pointer.Remote.Redundancy.MinReq+1))
+		return false, Error.Wrap(IrreparableError.New("segment %v cannot be repaired: only %d healthy pieces, %d required", path, numHealthy, pointer.Remote.Redundancy.MinReq+1))
```

Comment:
We should delete in the case of an `IrreparableError`. In fact, we have an explicit case for it in `repairer.go`.

Reply:
fixing